### PR TITLE
Add distributed PECOS README

### PR DESCRIPTION
**Pull request by weiliw-amz**
*Issue #, if available:*
N/A

*Description of changes:*
Added README for `pecos.distributed` modules.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pecos/distributed/xmc/xlinear/READMD.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+python3 -m pecos.xmc.xlinear.predict \
+-x ./xmc-base/eurlex-4k/tfidf-attnxml/X.tst.npz \
+-y ./xmc-base/eurlex-4k/Y.tst.npz \
+-m ./eurlex_4k_model
+```
+
+## Appendix: Distributed Efficiencies Q&A
+Extra distributed training parameters are available for better time and memory efficiency.
+
+**Q: What should I do if distributed model training is too slow, or consume too much memory on main node?**
```
Comment (jiong-zhang) [inline pecos/distributed/xmc/xlinear/READMD.md]: This is more like "why meta-clustering is too slow", rather than the distributed training. I'd rephrase this to avoid confusion.

File: pecos/distributed/xmc/xlinear/READMD.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+A: Use `PII` embedding for meta-tree label embedding to accelerate the clustering:
+```bash
+mpiexec -f hostfile -n ${NUM_MACHINE} \
+python3 -m pecos.distributed.xmc.xlinear.train \
+-x ${X_PATH} -y ${Y_PATH} -m ${MODEL_DIR} \
+--meta-label-embedding-method pii
+```
+This would produce slightly different models.
+
+**Q: What should I do if my model is too big, and MPI report buffer exceed limit error in training?**
```
Comment (jiong-zhang) [inline pecos/distributed/xmc/xlinear/READMD.md]: "model is too big" is too vague an issue. I'd recommend to paste the actual MPIBufferError here since the solution is particular for that error.

File: pecos/distributed/xmc/xlinear/READMD.md (hunk 1 above)
Comment (weiliw-amz) [reply pecos/distributed/xmc/xlinear/READMD.md]: I'm a little holding back for exposing `meta-clustering` level of info here because user may have no knowledge if we do not put a picture of the algorithm here, so I used a vague term.
But I'll change it for now. Later we could have a more detailed README for the algorithm.

**Review by jiong-zhang — APPROVED** [review]
